site-manage/pageorder/tool/src/java/org/sakaiproject/site/tool/helper/order/impl/SitePageEditHandler.java

```diff
@@ -512,22 +516,18 @@ public boolean allowDisable(SitePage page) {
         if (!(serverConfigurationService.getBoolean(DISABLE_ENABLED_CFG, true))) {
             return false;
         }
-        List<String> permissions = getRequiredPermissions(page);
+        List<String> permissions = getRequiredPermissions(page).stream().flatMap(Collection::stream).collect(Collectors.toList());
         return !(permissions.isEmpty() || permissions.contains(SITE_UPD) || permissions.contains(SITE_VISIT));
     }
 
-    private List<String> getRequiredPermissions(SitePage page) {
+    private List<Set<String>> getRequiredPermissions(SitePage page) {
+
         List<ToolConfiguration> tools = page.getTools();
         if (tools.size() == 1) {
-            // Only if there is one
-            ToolConfiguration toolConfiguration = tools.get(0);
-            String functions = toolConfiguration.getConfig().getProperty(TOOL_CFG_FUNCTIONS);
-            if (functions != null && functions.length() > 0) {
-                return new ArrayList<>(Arrays.asList(StringUtils.split(functions, ',')));
-            }
+            return toolManager.getRequiredPermissions(tools.get(0));
```

Comment:
The rest of the PR looks awesome bu I don't get this part. I love the rest of the changes but 

`getRequiredPermissions(SitePage page)` still seems confusing to me. What happens if the `tools.size() > 1`? Perhaps you're not changing the logic here though and this is the same as it was.